evm/src/cpu/kernel/interpreter.rs

```diff
@@ -120,17 +123,46 @@ pub(crate) fn run_interpreter_with_memory(
     Ok(interpreter)
 }
 
-pub(crate) fn run<'a>(
+pub(crate) fn run<'a, F: Field>(
     code: &'a [u8],
     initial_offset: usize,
     initial_stack: Vec<U256>,
     prover_inputs: &'a HashMap<usize, ProverInputFn>,
-) -> anyhow::Result<Interpreter<'a>> {
+) -> anyhow::Result<Interpreter<'a, F>> {
     let mut interpreter = Interpreter::new(code, initial_offset, initial_stack, prover_inputs);
     interpreter.run()?;
     Ok(interpreter)
 }
 
+/// Simulates the CPU execution from `state` until the program counter reaches `final_label`  
+/// in the current context.
+pub(crate) fn simulate_cpu_and_get_user_jumps<F: Field>(
+    final_label: &str,
+    state: &GenerationState<F>,
+) -> Option<HashMap<usize, Vec<usize>>> {
+    if state.jumpdest_table.is_some() {
+        None
+    } else {
+        let halt_pc = KERNEL.global_labels[final_label];
+        let initial_context = state.registers.context;
+        let mut interpreter =
+            Interpreter::new_with_state_and_halt_condition(state, halt_pc, initial_context);
+
+        log::debug!("Simulating CPU for jumpdest analysis.");
+
+        interpreter.run().unwrap();
+
+        log::debug!("jdt = {:?}", interpreter.jumpdest_table);
```

Comment:
I guess this can either be removed or put one level lower, to the `trace` level?

Reply:
additionally let's not use abbreviations
```suggestion
        log::debug!("Jumpdest table = {:?}", interpreter.jumpdest_table);
```

Reply:
This are not necessary. I just forgot to remove them.